lms/djangoapps/course_goals/api.py

```diff
@@ -22,17 +21,20 @@ def add_course_goal(user, course_id, goal_key):
         user: The user that is setting the goal
         course_id (string): The id for the course the goal refers to
         goal_key (string): The goal key for the new goal.
-
+        manually_updated (Boolean): Indicates if the goal was manually updated by the learner.
     """
     course_key = CourseKey.from_string(str(course_id))
     current_goal = get_course_goal(user, course_key)
     if current_goal:
         # If a course goal already exists, simply update it.
         current_goal.goal_key = goal_key
-        current_goal.save(update_fields=['goal_key'])
+        current_goal.manually_updated = manually_updated
+        current_goal.save(update_fields=['goal_key', 'manually_updated'])
     else:
         # Otherwise, create and save a new course goal.
-        new_goal = models.CourseGoal(user=user, course_key=course_key, goal_key=goal_key)
+        new_goal = CourseGoal(
+            user=user, course_key=course_key, goal_key=goal_key, manually_updated=manually_updated
+        )
         new_goal.save()
```

Comment:
This could be replaced with `CourseGoal.objects.update_or_create(user=user, course_key=course_key, defaults={goal, manually_updated})` yeah?

That wouldn't hit your new custom `save()` code though.